.buildkite/commands/configure-git-for-release-management.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -eu
+
+# Git command line client is not configured in Buildkite. Temporarily, we configure it in each step.
+# Later on, we should be able to configure the agent instead.
+curl -L https://api.github.com/meta | jq -r '.ssh_keys | .[]' | sed -e 's/^/github.com /' >> ~/.ssh/known_hosts
+git config --global user.email "mobile+wpmobilebot@automattic.com"
+git config --global user.name "Automattic Release Bot"
+
+# Buildkite is currently using the https url to checkout. We need to override it to be able to use the deploy key.
+git remote set-url origin git@github.com:wordpress-mobile/WordPress-Android.git
```

Comment:
💡  I wonder if it would be better to make this a bit more generic, like:
```bash
REPO_NO_TRAILING=${BUILDKITE_REPO%/}
git remote set-url origin  ${REPO_NO_TRAILING/#https:\/\/github.com\//git@github:}
```
Or alternatively:
```bash
REPO_SLUG=$(basename $(dirname "$BUILDKITE_REPO"))/$(basename "$BUILDKITE_REPO")
git remote set-url origin git@github.com:${REPO_SLUG}.git
```

Reply:
If this was in a toolkit repo, it'd make sense to go with a generic version. However, since this is part of WordPress-Android, I think this just makes it harder to understand the code.

Furthermore, it looks like I forgot to mention that this is a temporary issue specific to WordPress-Android. We should not be using the `https` url to clone in the first place. I just didn't have a chance to figure out why this is the case for WordPress-Android. I think we used to have a specific configuration for WPAndroid, but I couldn't find it yet.

For example, [we don't need this bit in `release-toolkit-demo`.](https://github.com/wordpress-mobile/release-toolkit-demo/blob/trunk/.buildkite/commands/configure-git.sh) I also checked [FluxC](https://buildkite.com/automattic/wordpress-fluxc-android/builds/2488#01881178-a932-4efc-976f-17dec5307aab/135-139) & [WCAndroid](https://buildkite.com/automattic/woocommerce-android/builds/12037#01881100-1de8-4f3a-92e2-dcc8ada0c193) and they both use the `ssh` url.

Reply:
Aaah.. and I think I might have found the culprit. All this time I've been checking the Buildkite settings' `Checkout using:` section which says that we are using `ssh` to checkout. However, I just checked the terraform and the `repository` field for WPAndroid is `https://github.com/wordpress-mobile/WordPress-Android/`. I compared this against WPiOS, WCAndroid & FluxC and they all use the `ssh` url. So, I think we'll be able to get rid of this configuration all together.

Having said that, I hope my previous logic also made sense in terms of keeping the specific url.

Reply:
> If this was in a toolkit repo, it'd make sense to go with a generic version. However, since this is part of WordPress-Android, I think this just makes it harder to understand the code.

🤦‍♂️ I've reviewed so many PRs in the past few days that I think I lost track of which reps each PR was about, and since I reviewed a PR about `buildkite-ci` repo just before this one, I think I didn't realize I was now reviewing a WPAndroid repo when I switched to this review 🤦‍♂️🤦  Sorry 😅

Reply:
No worries at all 🤗 

> I've reviewed so many PRs in the past few days

I noticed ❤️ - thank you for doing that! 🙇‍♂️